Assets/Mirage/Runtime/NetworkIdentity.cs

```diff
@@ -143,17 +145,29 @@ public sealed class NetworkIdentity : MonoBehaviour
         /// </summary>
         public bool HasAuthority { get; internal set; }
 
-        /// <summary>
-        /// The set of network connections (players) that can see this object.
-        /// </summary>
-        public readonly HashSet<INetworkPlayer> observers = new HashSet<INetworkPlayer>();
-
         /// <summary>
         /// Unique identifier for this particular object instance, used for tracking objects between networked clients and the server.
         /// <para>This is a unique identifier for this particular GameObject instance. Use it to track GameObjects between networked clients and the server.</para>
         /// </summary>
         public uint NetId { get; internal set; }
 
+        public IEnumerable<INetworkPlayer> Observers
+        {
+            get
+            {
+                if (ServerObjectManager == null)
+                    return Enumerable.Empty<INetworkPlayer>();
```

Comment:
is ServerObjectManager or interestManager being null an allowed state? should we be throwing here instead of returning empty?

Reply:
this happens on testing,  But to your question,  yes it is a valid state,  I can create an object with an NI,  and then I can call Observers without spawning it.   In my mind in that state we just don't have observers.